### PR TITLE
[Docs] Disable providing data

### PR DESCRIPTION
**Pull request by loic425**
| Q               | A
| --------------- | -----
| Bug fix?        | no (but missing documentation)
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets |
| License         | MIT

### CODE REVIEW

File: docs/providers.md [hunk 1]
```diff
@@ -115,15 +125,77 @@ declare(strict_types=1);
...
+
+```php
+// src/BoardGameBlog/Infrastructure/Sylius/State/Processor/DeleteBoardGameProcessor.php
+
+namespace App\BoardGameBlog\Infrastructure\Sylius\State\Processor;
+
+use Sylius\Resource\Context\Option\RequestOption;
+use Webmozart\Assert\Assert;
+
+final class DeleteBoardGameProcessor implements ProcessorInterface
```
Comment (diimpp) [inline docs/providers.md:183]: Missing `use` for ProcessorInterface, CommandBusInterface, Operation, Context, BoardGameResource, DeleteBoardGameCommand, BoardGameId

File: docs/providers.md [hunk 2]
```diff
@@ -3,10 +3,12 @@
 Providers retrieve data from your persistence layer.
 
 <!-- TOC -->
-* [Default providers](#default-providers)
-  * [Custom repository method](#custom-repository-method)
-  * [Custom repository arguments](#custom-repository-arguments)
-* [Custom providers](#custom-providers)
+* [Providers](#providers)
```
Comment (diimpp) [inline docs/providers.md]: Adding h1 header of the current page to the TOC looks a bit strange and doesn't match other doc files -- is it necessary? 

image

image

File: docs/processors.md [hunk 3]
```diff
@@ -120,6 +127,9 @@ namespace App\BoardGameBlog\Infrastructure\Sylius\Resource;
 final class BoardGameResource implements ResourceInterface
 ```
 
+Note that in that kind of operation, you can disable providing data.
```
Comment (diimpp) [inline docs/processors.md]: > Note that in that kind of operation, you can disable providing data.

What kind of operation? Would be cool to re-phrase a bit to be more specific

Comment (diimpp) [reply docs/processors.md]: Once I've read "Disable providing data" section, I've understood the point (delete, put, patch), but `note` phrasing needs to reflect it somehow.

File: docs/providers.md (hunk 1 above)
Comment (loic425) [reply docs/providers.md:183]: done

File: docs/processors.md (hunk 3 above)
Comment (loic425) [reply docs/processors.md]: I've replaced "in that kind of operation" with "in a delete operation".

File: docs/providers.md [hunk 4]
```diff
@@ -115,15 +124,84 @@ declare(strict_types=1);
...
+In some cases, you may want not to read data.
+
+For example, in a delete operation, you can implement your custom delete processor without reading it before.
+
+```php
+// src/BoardGameBlog/Infrastructure/Sylius/Resource/BoardGameResource.php
+
+declare(strict_types=1);
+
+namespace App\Entity;
```
Comment (GSadee) [inline docs/providers.md]: According to other example classes, shouldn't it be:
```suggestion
namespace App\BoardGameBlog\Infrastructure\Sylius\Resource;
```
🤔

**Review by GSadee — APPROVED** [review]

**Comment by GSadee** [comment]
Thanks, Loïc! :1st_place_medal:

**Pull request merged**